### DISCUSSION TITLE
Plugin to help maintaining focus

### DISCUSSION THREAD

**Discussion by Glyphack** [ActivityWatch/activitywatch#925] [Ideas]
I was thinking about starting a plugin that would assist in having more focused time. I'm still not sure what features are valuable here so creating this discussion to get more ideas from the community.

The basic thing to start with is to let user select a category(or categories) they want to spend time on. From that moment the new plugin will check your activity, and detect if you are loosing focus by:
- Checking if you are spending time on another category
- (What are other ways to track this?)

In the timer end plugin sends a notification to the user. Pomodoro is a popular technique in this space and plugin can start a break time for the user.

**Comment by 2e3s** [comment]
I tried to make an app with somewhat related functionality long ago, but later I tried Pomodoro and continuing to develop the app turned out to be quite useless. Because I always know myself very well and without the app if I'm working on what I should work on. And because it's much more productive to keep the work in uninterrupted intervals as Pomodoro suggests which by the way automatically counts the work, than develop something for unproductive broken work schedule.

Here's what I've done for counting my work with Pomodoro in KDE by the way: https://gist.github.com/2e3s/d9e6bfa7f4c59cd0307ecf39512c6cb3

The app could help if it could block everything but the needed resources and apps (power of will enforcer), but it's a huge challenge, completely unrelated to time tracking. It would be much easier to divide your environment manually (different KDE activities, virtual desktops, browsers, machines). Maybe the app could send a notification just as a reminder if you interrupt your work on the category, but would it be really useful to spend time on?

**Reply by Glyphack** [reply]
I gave this some more thought I think my issue with focus has to be fixed by fixing my working habits and not be integrated into the time tracker 😄
